pkgs/by-name/py/pyrefly/package.nix

```diff
@@ -1,41 +1,30 @@
 {
   lib,
-  python3,
-  fetchPypi,
-  versionCheckHook,
-  nix-update-script,
+  fetchFromGitHub,
   rustPlatform,
-  maturin,
+  versionCheckHook,
+  nix-update-script
 }:
-python3.pkgs.buildPythonApplication rec {
+rustPlatform.buildRustPackage (finalAttrs: {
   pname = "pyrefly";
-  version = "0.17.1";
-  pyproject = true;
-
-  # fetch from PyPI instead of GitHub, since source repo does not have Cargo.lock
-  src = fetchPypi {
-    inherit pname version;
-    hash = "sha256-w4ivRtmApXiXQT95GI4vvYBop7yxdbbkpW+YTyFtgXM=";
-  };
-
-  cargoDeps = rustPlatform.fetchCargoVendor {
-    inherit src;
-    hash = "sha256-Op5ueVkzZTiJ1zeBGVi8oeLcfSzXMYfk5zEg4OGyA5g=";
+  version = "0.20.2";
+  src = fetchFromGitHub {
+    owner = "facebook";
+    repo = "pyrefly";
+    tag = finalAttrs.version;
+    hash = "sha256-E3CPs3/c5/0VKeRFI6dNRj0xUoU9YBV1rZ1qFt4E2+U=";
   };
 
-  build-system = [ maturin ];
-
-  nativeBuildInputs = with rustPlatform; [
-    cargoSetupHook
-    maturinBuildHook
-  ];
+  buildAndTestSubdir = "pyrefly";
+  cargoHash = "sha256-EkGw7y5NtJ6Yd3DzL4z+81I0v5WyIwZ28klHXMqHjJc=";
 
-  nativeCheckInputs = [ versionCheckHook ];
+  nativeInstallCheckInputs = [ versionCheckHook ];
+  doInstallCheck = true;
+  RUSTC_BOOTSTRAP = true;
```

Comment:
`minor-nitpick` (can ignore): Other than what's already been said about `env`, add a little comment above the env var like there was previously, saying why it's necessary.